src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.SdkArchiveDiff/PathWithVersions.cs

```diff
@@ -0,0 +1,112 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Text;
+
+public static class PathWithVersions
+{
+    public const string VersionPlaceholder = "{VERSION}";
+
+    public static bool Equal(string path1, string path2)
+    {
+        if (path1 == path2)
+        {
+            return true;
+        }
+
+        ReadOnlySpan<char> directory = path1;
+        ReadOnlySpan<char> directory2 = path2;
+        while (TryGetPathLeaf(directory, out var root, out var directoryPart) && TryGetPathLeaf(directory2, out var root2, out var directoryPart2))
+        {
+            if (!ReplaceVersionString(directoryPart).SequenceEqual(ReplaceVersionString(directoryPart2)))
```

Comment:
This is one place where that version code in arcade may be helpful. It rips the version elements of the paths out to create  aka.ms links, which is what is happening here.